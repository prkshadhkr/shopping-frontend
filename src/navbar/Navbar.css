.nav-link.active {
  color: gray;
}