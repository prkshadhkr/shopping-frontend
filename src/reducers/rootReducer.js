import { combineReducers } from 'redux';
import products  from './products';
import reviews from './reviews';
import orders from './orders';

const rootReducer = combineReducers(
  { products, reviews, orders }
)

export default rootReducer;