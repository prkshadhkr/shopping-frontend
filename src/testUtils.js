import React from  'react';
import UserContext from './user/UserContext';

const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { UserProvider };