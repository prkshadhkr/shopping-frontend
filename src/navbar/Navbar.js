import React, { useContext } from 'react';
import { Link, NavLink }  from 'react-router-dom';
import UserContext from '../user/UserContext';
import './Navbar.css';
import { useSelector } from 'react-redux';
import { quantityTotal } from '../helpers/cartTotal';

const Navbar = ({ handleLogout }) => {
  const { user } = useContext(UserContext);
  const qty = useSelector(state => quantityTotal(state.products.items));
  const val = useSelector(state => state.products.cartValue);
  const unit = qty > 1 ? "items" : "item";
  
  const navbarLoddedIn = () => {
    return (
      <>
        <ul className="navbar-nav ml-auto">
          { 
            user.is_admin ?
            <li className="nav-item mr-4">
              <NavLink to="/products/new" className="nav-link">
                Add a Product
              </NavLink>
            </li>: ""
           }

          <li className="nav-item mr-4">
            <NavLink to="/products" className="nav-link">
              Products
            </NavLink>
          </li>
          <li className="nav-item mr-4">
            <NavLink to="/cart" className="nav-link">
              Cart
            </NavLink>
          </li>
          <li className="nav-item mr-4">
            <h6 className="navbar-text text-info">
              { qty } {unit}, Total: $ {val}
            </h6>
          </li>
          <li className="nav-item mr-4">
            <NavLink to="/Profile" className="nav-link">
              { user.username }
            </NavLink>
          </li>
          <li className="nav-item mr-4">
            <NavLink to="/" onClick={handleLogout} className="nav-link">
              Logout
            </NavLink>
          </li>
        </ul>
      </>
    )
  }

  const navbarLoggedOut = () => {
    return (
      <>
        <ul className="navbar-nav ml-auto">
          <li className="nav-item mr-4">
            <NavLink to="/login" className="nav-link">
              Login
            </NavLink>
          </li>
        </ul>
      </>
    )
  }
  return (
    <nav className="Navbar navbar navbar-expand-md">
      <Link to="/" className="navbar-brand">
        Shopping
      </Link>
      { user? navbarLoddedIn() : navbarLoggedOut() }
    </nav>
  )
}

export default Navbar;