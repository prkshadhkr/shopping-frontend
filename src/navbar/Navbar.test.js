import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Navbar from './Navbar';
import { UserProvider } from '../testUtils';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../reducers/rootReducer';

const store = createStore(rootReducer);

it ('render without crashing', function() {
  render (
    <MemoryRouter>
      <Provider store={store}>
        <UserProvider>
          <Navbar />
        </UserProvider>
      </Provider>
    </MemoryRouter>
  )
});

it('mathces the snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <Provider store={store}>
        <UserProvider>
          <Navbar />
        </UserProvider>
      </Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it ('matches the snapshot when logged out', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <Provider store={store}>
        <UserProvider user={null}>
          <Navbar />
        </UserProvider>
      </Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})