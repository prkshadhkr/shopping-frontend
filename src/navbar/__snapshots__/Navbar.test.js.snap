// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navbar navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Shopping
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`mathces the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navbar navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Shopping
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/products"
        >
          Products
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/orders"
        >
          Cart
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/orders/testing"
        >
          Order
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <h6
          class="navbar-text text-info"
        >
          0 item, Total: $ 0
        </h6>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/Profile"
        >
          testing
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Logout
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
